src/components/Landing/ConnectAndCreateButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { displayFrom, displayTo } from 'helpers/visibilityClassnames'
+import { handleError } from '@big-whale-labs/frontend-utils'
+import { useLocation } from 'wouter'
+import BurnerWalletStore from 'stores/BurnerWalletStore'
+import Button from 'components/ui/Button'
+import getErrorMessage from 'helpers/getErrorMessage'
+import walletStore from 'stores/WalletStore'
+
+const defaultMessage =
+  'We could not match the username with the wallet you have connected. Please review them and try again.'
+
+export default function ({
+  account,
+  loading,
+  burnerLoading,
+  onError,
+  onLoading,
+}: {
+  account?: string
+  loading: boolean
+  burnerLoading: boolean
+  onError: (error: string) => void
+  onLoading: (loading: boolean) => void
+}) {
+  const [, setLocation] = useLocation()
+
+  const createBurnerWallet = async () => {
+    onError('')
+    onLoading(true)
+
+    try {
+      if (!walletStore.account) await walletStore.connect(true)
+      if (!walletStore.account) return onError('Please connect the wallet')
+      await BurnerWalletStore.generateBurnerWallet(walletStore.account)
+      walletStore.exit()
+      setLocation('/wallet')
+    } catch (error) {
+      const errorMessage = getErrorMessage(error)
+      onError(typeof errorMessage === 'string' ? errorMessage : defaultMessage)
+      handleError(error)
+    } finally {
+      onLoading(false)
+    }
+  }
+
+  return (
+    <>
+      <div className={displayTo('md')}>
+        <Button
+          small
+          center
+          fullWidth
+          loadingOverflow
+          type="primary"
+          loading={loading || burnerLoading}
+          onClick={createBurnerWallet}
+        >
+          {account ? 'Create burner wallet' : 'Connect & create burner wallet'}
```

Comment:
The text and condition repeats, we should extract it into the `const`

Reply:
Did